app/src/main/java/com/duckduckgo/app/autocomplete/api/AutoComplete.kt

```diff
@@ -75,12 +79,73 @@ class AutoCompleteApi @Inject constructor(
             .toObservable()
 
     private fun getAutoCompleteBookmarkResults(query: String) =
-        bookmarksDao.bookmarksByQuery("%$query%")
+        bookmarksDao.bookmarksObservable()
+            .map { rankBookmarks(query, it) }
             .flattenAsObservable { it }
             .map {
-                AutoCompleteBookmarkSuggestion(phrase = it.url, title = it.title ?: "", url = it.url)
+                AutoCompleteBookmarkSuggestion(phrase = it.url.toUri().resourceNameAndPath(), title = it.title.orEmpty(), url = it.url)
```

Comment:
Just a small issue with this. If you try to save a bookmark for a URL with query parameters they won't be used when editing the query. For example if you go to duckduckgo.com where by default we add some query parameters and save that as bookmark, those query parameters are not re-added when trying to edit the query.